### PR TITLE
Fix #16614 - Correct the dataseek offset on MySQLI extension

### LINKED ISSUE #16614: Argument #1 ($offset) must be greater than or equal to 0
5.1.0-rc1
PHP 8.0.1

When trying to browse an empty table:

```
PHP message: PHP Fatal error:  Uncaught ValueError: mysqli_result::data_seek(): Argument #1 ($offset) must be greater than or equal to 0 in /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Dbal/DbiMysqli.php:308
Stack trace:
#0 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Dbal/DbiMysqli.php(308): mysqli_result->data_seek()
#1 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/DatabaseInterface.php(2036): PhpMyAdmin\Dbal\DbiMysqli->dataSeek()
#2 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Display/Results.php(4339): PhpMyAdmin\DatabaseInterface->dataSeek()
#3 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Display/Results.php(4070): PhpMyAdmin\Display\Results->getSortedColumnMessage()
#4 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Sql.php(1356): PhpMyAdmin\Display\Results->getTable()
#5 /srv/phpMyAdmin-5.1+snapshot-english/libraries/classes/Sql.php(1157): PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable()
#6 /srv/phpMyAdmin-5.1+snapshot-english/librarie
```

### PR DESCRIPTION
**Pull request by williamdes**
Fixes #16614

### CODE REVIEW

**Comment by OlafvdSpek** [comment]
It masks the real problem, I wouldn't do it.

**Comment by williamdes** [comment]
@mauriciofauth what are your thoughts on this PHP 8 fix ?

**Comment by ibennetch** [comment]
I'm also quite curious why $offset would be less than 0.

On the surface this looks acceptable to me, but I'd really like to know more about what's gone wrong elsewhere in the code to call this.

**Comment by williamdes** [comment]
> I'm also quite curious why $offset would be less than 0.
> 
> On the surface this looks acceptable to me, but I'd really like to know more about what's gone wrong elsewhere in the code to call this.

I did not check but we have operations like ` $this->properties['num_rows'] - 1` and if rows = 0 then 0-1 = -1

**Comment by ibennetch** [comment]
Yeah, that makes sense. I think I'm okay with this. What one person considers masking the problem, another can consider "dealing with it in another code block" and it seems that any time we have a negative offset, it's meant to be zero anyway. I'm also curious what @mauriciofauth has to say but I would be okay with this.

**Comment by williamdes** [comment]
@sudo-robot deploy

**Comment by OlafvdSpek** [comment]
> Yeah, that makes sense. I think I'm okay with this. What one person considers masking the problem, another can consider "dealing with it in another code block" and it seems that any time we have a negative offset, it's meant to be zero anyway. I'm also curious what @mauriciofauth has to say but I would be okay with this.

Suppose I call dataSeek(res, -2). Without this PR, the bug would be caught and I could fix it. With this PR, the bug wouldn't be caught and I can't be sure I really meant dataSeek(res, 0). Maye I meant dataSeek(res, 2)..

File: libraries/classes/Dbal/DbiMysqli.php [hunk 1]
```diff
@@ -305,7 +305,7 @@ public function fetchRow($result): ?array
      */
     public function dataSeek($result, $offset)
     {
-        return $result->data_seek($offset);
+        return $result->data_seek($offset < 0 ? 0 : $offset);
```
Comment (mauriciofauth) [inline libraries/classes/Dbal/DbiMysqli.php:308]: A better solution is to check for this earlier. For example, we can check if `$this->properties['num_rows'] > 0`.
```php
$dbi->dataSeek($dt_result, $this->properties['num_rows'] - 1);
```

Comment (williamdes) [reply libraries/classes/Dbal/DbiMysqli.php]: Indeed, I did that now

**Review by MauricioFauth — APPROVED** [review]

**Comment by ibennetch** [comment]
👍

**Pull request merged**